src/internal/admindb/admindb.go

```diff
@@ -0,0 +1,64 @@
+// Package admindb provides functions for interacting with the admin.* database schema.
+package admindb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jmoiron/sqlx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+)
+
+// ScheduleRestart schedules every Pachyderm pod to restart at `when`.
+func ScheduleRestart(ctx context.Context, tx *pachsql.Tx, when time.Time, reason, who string) error {
+	if _, err := tx.ExecContext(ctx, `insert into admin.restarts (restart_at, reason, created_by) values ($1, $2, $3)`, when, reason, who); err != nil {
+		return errors.Wrap(err, "insert")
+	}
+	return nil
+}
+
+type restartMessage struct {
+	Reason    string `db:"reason"`
+	CreatedBy string `db:"created_by"`
+}
+
+func (m restartMessage) String() string {
+	var msg strings.Builder
+	if m.Reason != "" {
+		msg.WriteString(m.Reason)
+	} else {
+		msg.WriteString("restart requested")
+	}
+	if m.CreatedBy != "" {
+		fmt.Fprintf(&msg, " by %v", m.CreatedBy)
+	}
+	return msg.String()
+}
+
+func message(messages ...restartMessage) string {
+	parts := make([]string, len(messages))
+	for i, m := range messages {
+		parts[i] = m.String()
+	}
+	return strings.Join(parts, "; ")
+}
+
+// ShouldRestart returns true (and the reason) if a pod that started at `startup` should restart.  It also cleans up outdated restart requests (90 days old).
```

Comment:
Done.  We adjust the startup time based on the database time.  Because of querying latency, this makes it look like the pod started up EARLIER than it actually did, which is what we want.  We also log the time, because it will probably be interesting in a debug dump.